sdk-java/src/main/java/io/littlehorse/sdk/wfsdk/internal/WorkflowThreadImpl.java

```diff
@@ -347,10 +344,12 @@ public void doIfElse(WorkflowCondition condition, IfElseBody ifBody, IfElseBody
         lastNodeCondition = cond.getSpec();
         ifBody.body(this);
 
-        // Close off the tree. The bottom node from the ifBlock tree is also
-        // going to be the bottom node from the elseBlock.
-        addNopNode();
-        String joinerNodeName = lastNodeName;
+        String lastNodeFromIfBlockName = lastNodeName;
+        Node.Builder lastNodeFromIfBlock = spec.getNodesOrThrow(lastNodeName).toBuilder();
+        List<VariableMutation> variablesFromIfBlock = new ArrayList<>(variableMutations.size());
```

Comment:
From line 349 to 352 could be extracted to a method like how I did it on the Python SDK